commands/utility/clearinfractions.js

```diff
@@ -0,0 +1,111 @@
+const Discord = require('discord.js');
+const dateFormat = require('dateformat');
+
+module.exports = {
+  name: 'clearinfractions',
+  description: 'Removes all infractions from a user',
+
+  execute(msg, con, args) {
+    const {status, err, userInfraction} = validInfraction(msg, args);
+    if (!status) {
+      return msg.reply(err);
+    }
+
+    findInfractions(msg, userInfraction, args, con);
+  },
+};
+
+function validInfraction(msg, args) {
+  const data = {
+    status: false,
+    err: null,
+    userInfraction: null,
+  };
+
+  if (!msg.member.roles.cache.some((role) => role.name === 'Admin')) {
+    data.err = 'You must be an Admin to use this command.';
+    console.log(data);
+    return data;
+  }
+
+  data.userInfraction =
+    msg.mentions.members.first() || msg.guild.members.cache.get(args[0]);
+  if (!data.userInfraction) {
+    data.err = 'Please provide a user to remove infractions.';
+    return data;
+  }
+
+  data.status = true;
+  return data;
+}
+
+function findInfractions(msg, userInfraction, args, con) {
+  const sql = `SELECT COUNT(id) FROM infractions WHERE user = ? AND valid = ?;`;
+
+  const values = [userInfraction.id, true];
+
+  const escaped = con.format(sql, values);
+
+  con.query(escaped, function (err, result) {
+    if (err) {
+      console.log(err);
+    } else {
+      if (result[0].count > 0) {
+        clearInfractionSQL(msg, userInfraction, args, con);
+      } else {
+        msg.reply('This user does not have any infractions');
+      }
```

Comment:
This block will never satisfy the `if` condition, because `result[0].count` is `undefined`.  
`result` looks something like this
```
[ RowDataPacket { 'COUNT(id)': 2 } ]
```

So it looks like it should be
```javascript
if (result[0]['COUNT(id)'] > 0) {}
```